src/mame/drivers/vsmjtria.cpp

```diff
@@ -447,6 +452,13 @@ static INPUT_PORTS_START( vsmjtria )
 	PORT_DIPNAME( 0x80, 0x80, "Unknown 4-7" )
 	PORT_DIPSETTING(    0x80, DEF_STR( Off ) )
 	PORT_DIPSETTING(    0x00, DEF_STR( On ) )
+
+	// single slider switch, not read by CPU
+	// grounds NVRAM contents, simulated in machine_reset fn
+	PORT_START("SW2")
+	PORT_DIPNAME( 0x01, 0x00, "Clear NVRAM at boot" )
+	PORT_DIPSETTING(    0x00, DEF_STR( No ) )
+	PORT_DIPSETTING(    0x01, DEF_STR( Yes ) )
```

Comment:
Does it just ground the backup power to the NVRAM, or does it completely disconnect power from it even when the board is powered up from the main edge connector?  If the former, then clearing memory on reset is reasonable, but if the latter, then it should black-hole writes when it’s on.

Also, should this be `IPT_MEMORY_RESET` with `PORT_TOGGLE`?  We have a specific input type for memory reset (yeah, I realise the lack of feedback on toggle inputs is an annoying limitation of MAME right now).

Reply:
> yeah just metered it, with sw1 in the normal position 5v comes from the pcb or battery, standard setup for battery backup srams
> diodes to prevent the battery powering the rest of the pcb
> flip the switch and the power input for the two ram is grounded - boom

I honestly dislike the IPT_MEMORY_RESET approach here for the reason you described, this is also technically still a dip switch at heart (i.e. eventually requiring `DIPLOCATION`)

Reply:
No it technically *isn’t* a DIP switch – DIP stands for Dual Inline Package, i.e. a form factor with two rows of pins.  DIP switches are multiple single-pole single-throw switches in a dual inline package (usually miniature slider switches, but occasionally miniature rocker switches, or miniature toggle switches may be used in right-angle DIP switches in some rare cases).

SW2 is a PCB-mounted single-pole dual-throw slider switch (common terminal in the centre).  Right now, MAME doesn’t give a great deal of choice for controls.  There are user-facing toggle controls, DIP switches, then machine configuration for everything else.

Using a DIP location for something that isn’t a DIP switch is kind of misleading, although various things do it, e.g. because MAME doesn’t have a way to give jumper locations (speaking of which, a proper jumper description language is going to be difficult if anyone ever tries to implement it, given the number of different arrangements they can have).

In this case, it could be named “Memory Reset (SW2)” or something to include the silkscreen label.

Reply:
My point is that it belongs _more_ to the dipswitch menu rather than the machine configuration.